extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -59,27 +62,40 @@ export abstract class Validation {
 		return this._valueGetter();
 	}
 
-	protected getVariableValue(variable: string): Promise<ValidationValueType> {
-		return this._variableValueGetter!(variable);
+	protected getTargetValue(variable: string): Promise<ValidationValueType> {
+		return this._targetValueGetter!(variable);
 	}
 
-	constructor(validation: ValidationInfo, protected _valueGetter: ValueGetter, protected _variableValueGetter?: VariableValueGetter) {
+	protected async onValidation(isValid: boolean): Promise<void> {
+		await this._onValidation(isValid);
+	}
+
+	constructor(validation: ValidationInfo, protected _onValidation: onValidation, protected _valueGetter: ValueGetter, protected _targetValueGetter?: TargetValueGetter, protected _onTargetValidityChangedGetter?: OnTargetValidityChangedGetter, protected _onNewDisposableCreated?: (disposable: vscode.Disposable) => void) {
 		this._description = validation.description;
 	}
+
+	protected isUndefinedOrEmpty(value: ValidationValueType): boolean {
+		return value === undefined || (typeof value === 'string' && value.length === 0);
```

Comment:
This can just be put as a standalone helper function in the utils module

Reply:
Ping

Reply:
moved isUndefinedOrEmpty to utils